File: Data Structures & Algorithms/is-subsequence/submission-4.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        if len(s) > len(t):
            return False
        
        if len(s) == 0:
            return True

        sp = 0
        tp = 0
        res = ""
        while tp < len(t):
            if s[sp] == t[tp]:
                res += t[tp]
                if res == s:
                    return True
                else:
                    tp += 1
                    sp += 1
            else:
                tp += 1
        return False

File: Data Structures & Algorithms/reverse-integer/submission-4.rs
impl Solution {
    pub fn reverse(mut x: i32) -> i32 {
        let mut res = 0i32;

        while x != 0 {
            let Some(next) = res.checked_mul(10).and_then(|n| n.checked_add(x % 10)) else {
                return 0;
            };

            res = next;
            x /= 10
        }

        res
    }
}
